source/common/common/callback_impl.h

```diff
@@ -50,6 +51,10 @@ template <typename... CallbackArgs> class CallbackManager {
 
     CallbackManager& parent_;
     Callback cb_;
+
+    // the iterator of this callback holder inside callbacks_ list
+    // upon removal, use this iterator to delete callback holder in O(1)
+    typename std::list<CallbackHolder>::iterator it_;
```

Comment:
have you considered to use a hashmap which may make the code a little easier to read?

Reply:
I am worried that a hashmap may incur more memory consumption and performance penalty. Also, storing a list iterator make the code verbose without all the hashing and searching. As suggested, I have added some comment to make the code more readable.

Reply:
I think what is done here is fine, as it's basically just `std::list` iterator semantics, which could be explained a bit better but are fairly clear.